File: Pet-Adoption-Shelter-Front/src/apis/admin/admin.api.ts
import type { VeiwAllUsersResponse, ViewUserProfile } from "@/types/admin/admin.dto";
import { privateApi } from "../common/axiosInstance";
import type { ApiResponse } from "@/types/common/ApiResponse";
import { ADMIN_PATH } from "./admin.path";

export const adminApi = {
  getAllUsers: async (): Promise<VeiwAllUsersResponse> => {
    const res = await privateApi.get<ApiResponse<VeiwAllUsersResponse>>(ADMIN_PATH.LIST);
    return res.data.data;
  },
  getUserProfile: async (userId: number): Promise<ViewUserProfile> => {
    const res = await privateApi.get<ApiResponse<ViewUserProfile>>(ADMIN_PATH.BY_ID(userId));
    return res.data.data;
  }
}

File: mini-proj-Pet-Adoption-Shelter/src/main/java/org/example/miniprojpetadoptionshelter/common/apis/AdminApi.java
package org.example.miniprojpetadoptionshelter.common.apis;

public class AdminApi {
    private AdminApi() {};

    public static final String ROOT = ApiBase.BASE + "/admin";
    public static final String LIST = ROOT + "/users";
    public static final String BY_ID = ROOT + "/users/{userId}";
}
